src/meta/processors/partsMan/CreateSpaceProcessor.cpp

```diff
@@ -174,20 +189,42 @@ CreateSpaceProcessor::pickHosts(PartitionID partId,
     return pickedHosts;
 }
 
-std::vector<HostAddr>
+StatusOr<std::unordered_map<HostAddr, int32_t>>
+CreateSpaceProcessor::getHostLoading() {
+    std::unique_ptr<kvstore::KVIterator> iter;
+    const auto& prefix = MetaServiceUtils::partPrefix();
+    auto code = kvstore_->prefix(kDefaultSpaceId, kDefaultPartId, prefix, &iter);
+    if (code != kvstore::ResultCode::SUCCEEDED) {
+        LOG(ERROR) << "List Parts Failed";
+        return Status::Error("List Parts Failed");
+    }
+
+    std::unordered_map<HostAddr, int32_t> result;
+    while (iter->valid()) {
+        auto hosts = MetaServiceUtils::parsePartVal(iter->val());
+        for (auto& host : hosts) {
+            result[host]++;
+        }
+        iter->next();
+    }
+    return result;
+}
+
+StatusOr<std::vector<HostAddr>>
 CreateSpaceProcessor::pickHostsWithZone(PartitionID partId,
                                         const std::vector<std::string>& zones,
-                                        int32_t replicaFactor) {
+                                        int32_t replicaFactor,
+                                        std::unordered_map<HostAddr, int32_t>& loading) {
     auto startIndex = partId;
     std::vector<HostAddr> pickedHosts;
     for (int32_t i = 0; i < replicaFactor; i++) {
-        auto zoneName = zones[startIndex++ % zones.size()];
+        auto& zoneName = zones[startIndex++ % zones.size()];
```

Comment:
Is this right? If I understand correctly, for example, a group contains 4 zones, and we keep creating space of replica 3, they will be always on first 3 zones.